router.go

```diff
@@ -70,8 +70,9 @@ func (route *Route) addHandlerTo(routeStr string, h *handler) {
 		r := rs[i]
 		if _, ok := next[r]; !ok {
 			next[r] = NewRoute()
-			i++
 		}
+		// increase i whether create new route or not
+		i++
 		if i != len(rs) {
 			next = next[r].Children
 		}
```

Comment:
Do you test this yet?
```go
for next := route.Children, i := 0; i < len(rs); i++ {
	r := rs[i]
	if _, ok := next[r]; !ok {
		next[r] = NewRoute()
	}
	if i != len(rs) {
		next = next[r].Children
	}
}
```

Reply:
I change whole algorithm  at 90b558bf5034b47a6df4f2b94c6aa6efa57126b6